org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/RecordsFilter.java

```diff
@@ -58,6 +63,40 @@ boolean isHashCode(final MethodNode m) {
 			return cursor != null;
 		}
 
+		public static final List<Integer> ret = Arrays.asList(Opcodes.IRETURN,
+				Opcodes.LRETURN, Opcodes.FRETURN, Opcodes.DRETURN,
+				Opcodes.ARETURN);
```

Comment:
Can this be a set?

Reply:
Of course!

Reply:
As filters are performance critical we avoid out boxing at all. So neither List nor Set should be used for `int` values, just simple chain of comparisons and with or.